Add helpers for AMAUAT functions

ross-spencer opened the pull request:
Make it as easy as possible for developers to run automated tests via
this Makefile.

We add two new black-box helpers, one with docker, and one without.

We can also output a list of common tests to run. At the moment there
are four suggested tests.

Connected to archivematica/issues#793

ross-spencer commented:
Hi @replaceafill I am wondering if you have a view on these new helper functions for the `Makefile`. They are:
```
   * test-at-black-box-docker
   * test-at-black-box-local
   * test-at-list
```
I find them useful, I feel like they might help folks be more inclined to run the `black-box` work outside of Artefactual. But IDK. Interested to hear back from you, or anyone else following along that might have some input?

sevein commented:
What's the use case in `test-at-black-box-local`?
It would be the only rule doing non-docker related work.

`test-at-black-box-docker` could pass tags to `test-at-behave` so you don't have to duplicate the whole thing, e.g.:

```make
git-checkout:
	git checkout $(BRANCH)

git-checkout-master: BRANCH=master
git-checkout-master: git-checkout
```

`test-at-list` could be named as `test-at-help` and use [echoing](https://www.gnu.org/software/make/manual/html_node/Echoing.html#Echoing)?

ross-spencer commented:
> What's the use case in test-at-black-box-local?

I'm not wedded to keeping it. I didn't realize the `Makefile` in this project needed to remain docker only. It pulls together a few projects and so it works well as a general purpose SDK. We can develop the AMAUAT suite from here too. That being said, it's a valid question. I personally adopt the non-docker approach in my testing of the black-box scenarios. I've modified it to allow Py3 and Py2 arguments. Though as it turns out it looks like Py2 doesn't work at the moment so I feel there has been a shift there. I also think it's a useful recipe to include in the `Makefile` for anyone who might try it out. Not the strongest of reasons perhaps. One slight advantage is that you don't have to run the build command to reload the acceptance-test source before testing in Docker. It's not a huge deal to remove, but I wanted it to be part of the discussion.

> `test-at-black-box-docker` could pass tags to test-at-behave so you don't have to duplicate the whole thing, e.g.:

> `test-at-list` could be named as test-at-help and use echoing?

Yep, two commands I grappled with and tried to do first time around. I couldn't quite find the right invocation for either but your pointers helped, thanks. I've modified the script to remove the redundancy now. It looks a lot better!

sevein commented on compose/Makefile:
```diff
@@ -175,6 +200,27 @@ test-at-behave: test-at-build  ## AMAUAT: run behave, default is `make test-at-b
+	       --tags=black-box && \
+	cd "../../compose/" && \
+	deactivate && \
+	rm -r "venv"
```
You could leave the env in a location that is not tracked by git so we don't have to recreate on each run. I haven't tested but you could try something like the following:

```Makefile
PY ?= python3
TAGS ?= black-box
ENV ?= $(CURDIR)/.envs.ignoredir/archivematica-acceptance-tests
SRC ?= $(CURDIR)/../src/archivematica-acceptance-tests
test-at-behave-local:  ## Help message... (TAGS, PY)
	@virtualenv --quiet --python=$(PY) $(ENV)
	@pip install --quiet -r $(SRC)/requirements.txt
	$(ENV)/bin/python -m behave \
		--tags=$(TAGS) --other-flags
```

sevein replied:
Also to be considered, `virtualenv` would need to be listed in the [requirements section](https://github.com/artefactual-labs/am/tree/master/compose#requirements).

sevein commented on compose/Makefile:
```diff
@@ -164,6 +164,33 @@ test-at-check: test-at-build  ## AMAUAT: test browsers.
+   The most effective way to run these tests is to run them by tag. For
+   example:
+
+      $ make test-at-behave TAGS=black-box BROWSER=Firefox
```
I wonder why you chose this example instead of one with a tag where BROWSER is relevant.
A second example showing `test-at-black-box` would be nice.

sevein commented on compose/Makefile:
```diff
@@ -164,6 +164,33 @@ test-at-check: test-at-build  ## AMAUAT: test browsers.
+
+   Commonly used acceptance tests in the Archivematica suite:
+
+      * aip-encrypt 	   :Tests the encryption of AIPs.
```
Indentation issue here.

ross-spencer replied:
Can you elaborate on the indentation issue? This is how I wrote it.

sevein replied:
There's one tab there messing with my terminal.

![image](https://user-images.githubusercontent.com/606459/65992627-65e61f80-e444-11e9-9e87-4a1b7fb57c12.png)

![image](https://user-images.githubusercontent.com/606459/65992542-3fc07f80-e444-11e9-82e3-f2577d7dd06d.png)
